runtime/runtime/src/system.rs

```diff
@@ -64,16 +64,26 @@ pub fn staking(
     sender: &mut Account,
     validator_proposals: &mut Vec<ValidatorStake>,
 ) -> Result<Vec<ReceiptTransaction>, String> {
-    if sender.amount >= body.amount {
```

Comment:
add a check for 
`if body.amount < sender.staked` { return Err(format!("Trying to stake {} when only have {}")) }`